### PR TITLE
Adding typing to PPCA module

### PR DESCRIPTION
**Pull request by austinTalbot7241993**
What the title says + formatting some files.

### CODE REVIEW

File: python/python/bystro/supervised_ppca/gf_dropout_pt.py [hunk 1]
```diff
@@ -106,13 +108,13 @@ class PPCADropout(PPCA):
...
-        n_components=2,
-        n_supervised=1,
-        prior_options=None,
-        mu=1.0,
-        gamma=10.0,
-        delta=5.0,
-        training_options=None,
+        n_components:int=2,
+        n_supervised:int=1,
+        prior_options:Optional[dict]=None,
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/gf_dropout_pt.py:113]: Let’s use dict | None

https://docs.python.org/3/library/stdtypes.html#types-union

Less typing

**Review by akotlar — CHANGES_REQUESTED** [review]
Let’s use the alternative syntax for Optional, and drop the # type: ignore from the torch import

File: python/python/bystro/supervised_ppca/_base.py [hunk 2]
```diff
@@ -227,7 +234,11 @@ def transform_subset(
         return S
 
     def conditional_score(
-        self, X, observed_feature_idxs, weights=None, sherman_woodbury=False
+        self,
+        X: NDArray,
+        observed_feature_idxs: NDArray,
+        weights: Optional[NDArray] = None,
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/_base.py:240]: weights: NDArray | None = None

File: python/python/bystro/supervised_ppca/_base.py [hunk 3]
```diff
@@ -300,7 +315,11 @@ def conditional_score_samples(
         return scores
 
     def marginal_score(
-        self, X, observed_feature_idxs, weights=None, sherman_woodbury=False
+        self,
+        X: NDArray,
+        observed_feature_idxs: NDArray,
+        weights: Optional[NDArray] = None,
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/_base.py:321]: NDArray | None = None

File: python/python/bystro/supervised_ppca/_base.py [hunk 4]
```diff
@@ -366,7 +388,12 @@ def marginal_score_samples(
             )
         return scores
 
-    def score(self, X, weights=None, sherman_woodbury=False):
+    def score(
+        self,
+        X: NDArray,
+        weights: Optional[NDArray] = None,
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/_base.py:394]: NDArray | None = None

File: python/python/bystro/supervised_ppca/gf_dropout_pt.py [hunk 5]
```diff
@@ -31,11 +31,13 @@
...
 """
 import numpy as np
+from numpy.typing import NDArray
+from typing import Dict, List, Optional
 
 from tqdm import trange
-import torch # type: ignore
-from torch import nn # type: ignore
-from torch.distributions.multivariate_normal import MultivariateNormal # type: ignore
+import torch  # type: ignore
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/gf_dropout_pt.py:38]: let's get rid of the # type: ignore here and below, none of these need type : ignore

File: python/python/bystro/supervised_ppca/gf_dropout_pt.py [hunk 6]
```diff
@@ -106,13 +108,13 @@ class PPCADropout(PPCA):
...
-        gamma=10.0,
-        delta=5.0,
-        training_options=None,
+        n_components:int=2,
+        n_supervised:int=1,
+        prior_options:Optional[dict]=None,
+        mu:float=1.0,
+        gamma:float=10.0,
+        delta:float=5.0,
+        training_options:Optional[dict]=None,
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/gf_dropout_pt.py:117]: dict | None = None

File: python/python/bystro/supervised_ppca/gf_dropout_pt.py [hunk 7]
```diff
@@ -138,10 +140,10 @@ def fit(self, X, y, task="classification", progress_bar=True, seed=2021):
 
         Parameters
         ----------
-        X : np.array-like,(n_samples,n_covariates)
+        X : NDArray,(n_samples,n_covariates)
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/gf_dropout_pt.py:143]: `fit(self, X: NDArray, y: NDArray, task: str = "classification", progress_bar: bool = True, seed: int = 2021):` ; it may be that the task, progress_bar, seed are not needed because the type system infers them from the default values, if so, don't add those

File: python/python/bystro/supervised_ppca/gf_dropout_pt.py [hunk 8]
```diff
@@ -243,7 +245,7 @@ def fit(self, X, y, task="classification", progress_bar=True, seed=2021):
         self._store_instance_variables(trainable_variables)
         return self
 
-    def _store_instance_variables(self, trainable_variables):
+    def _store_instance_variables(self, trainable_variables: List):
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/gf_dropout_pt.py:248]: instead of `List` just use `list` and remove the List import

File: python/python/bystro/supervised_ppca/gf_generative_pt.py [hunk 9]
```diff
@@ -39,21 +39,26 @@
...
+from numpy.typing import NDArray
+from typing import Dict, List, Optional
 
 from sklearn.decomposition import PCA  # type: ignore
 from tqdm import trange  # type: ignore
-import torch # type: ignore
-from torch import nn # type: ignore
-from torch.distributions.multivariate_normal import MultivariateNormal # type: ignore
-from torch.distributions.gamma import Gamma # type: ignore
+import torch  # type: ignore
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/gf_generative_pt.py:47]: let's remove the type ignores on all of these

File: python/python/bystro/supervised_ppca/gf_generative_pt.py [hunk 10]
```diff
@@ -39,21 +39,26 @@
...
 from bystro.supervised_ppca._misc_np import softplus_inverse_np
 from bystro.supervised_ppca._base import BasePCASGDModel
 
 
 class PPCA(BasePCASGDModel):
     def __init__(
-        self, n_components=2, prior_options=None, training_options=None
+        self,
+        n_components: int = 2,
+        prior_options: Optional[dict] = None,
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/gf_generative_pt.py:60]: dict | None here and below

File: python/python/bystro/supervised_ppca/gf_generative_pt.py [hunk 11]
```diff
@@ -39,21 +39,26 @@
 None
 """
 import numpy as np
+from numpy.typing import NDArray
+from typing import Dict, List, Optional
```
Comment (akotlar) [inline python/python/bystro/supervised_ppca/gf_generative_pt.py:43]: you can remove this line. Instead of using Dict, List, and Optional types, use:

`dict`, `list`, and `T | None`

**Comment by akotlar** [comment]
fixes pr'd in https://github.com/austinTalbot7241993/bystro/pull/6/files

**Review by akotlar — APPROVED** [review]
